runtime/onert/core/include/ir/Model.h

```diff
@@ -175,8 +175,26 @@ class Model
 
 private:
   std::shared_ptr<backend::custom::IKernelBuilder> _kernel_builder;
-};
 
+public:
+  void add_metadata(const std::string &name, std::shared_ptr<const ir::Data> data)
+  {
+    _metadatas.emplace(name, data);
+  }
+
+  bool is_metadata_exist(const std::string &name) const
+  {
+    return _metadatas.find(name) != _metadatas.end();
+  }
+
+  std::shared_ptr<const ir::Data> get_metadata(const std::string name) const
+  {
+    return _metadatas.at(name);
+  }
+
+private:
+  std::unordered_map<std::string, std::shared_ptr<const ir::Data>> _metadatas;
```

Comment:
Let's make a Model to hold metadata-name and related buffer(Data). 

I have an plan to parse this data like this : (draft code - https://github.com/Samsung/ONE/pull/12152/files#r1426184845 )

Reply:
Just my curiosity: I didn't see `datas` often before. it looked a bit, perhaps data is already plural form. Anyway, s after `data` helps me that it is a collection of `metadata`. : )

Reply:
I've also thought about `metadatas`.
For me, `Metadata` looks like one word (not `meta` and `data`), and I'd like to emphasize a 'collection' of metadata.
So, As you guessed, I tried to use `metadatas`, despite it is grammarly a bit wrong.